contracts/multicall/GuardedMulticaller2.sol

```diff
@@ -0,0 +1,240 @@
+
+    /// @dev Error thrown when reference is invalid
+    error InvalidReference(bytes32 _reference);
+
+    /// @dev Error thrown when reference has already been executed
+    error ReusedReference(bytes32 _reference);
+
+    /// @dev Error thrown when deadline is expired
+    error Expired(uint256 _deadline);
+
+    /// @dev Error thrown when signer is not authorized
+    error UnauthorizedSigner(address _multicallSigner);
+
+    /// @dev Error thrown when signature is invalid
+    error UnauthorizedSignature(bytes _signature);
+
+    /// @dev Error thrown when call array is empty
+    error EmptyCallArray();
+
+    /// @dev Error thrown when call reverts
+    error FailedCall(Call _call);
+
+    /// @dev Error thrown when target address is not a contract
+    error NonContractAddress(Call _call);
+
+    /// @dev Error thrown when function signature is invalid
+    error InvalidFunctionSignature(Call _call);
+
+    /**
+     *
+     * @notice Grants DEFAULT_ADMIN_ROLE to the contract creator
+     * @param _owner Owner of the contract
+     * @param _name Name of the contract
+     * @param _version Version of the contract
+     */
+    // solhint-disable-next-line no-unused-vars
+    constructor(address _owner, string memory _name, string memory _version) EIP712(_name, _version) {
+        _grantRole(DEFAULT_ADMIN_ROLE, _owner);
+    }
+
+    /**
+     *
+     * @notice Execute a list of calls. Returned data from calls are ignored.
+     *  The signature must be generated by an address with MULTICALL_SIGNER_ROLE
+     *  The signature must be valid
+     *  The signature must not be expired
+     *  The reference must be unique
+     *  The reference must not be executed before
+     *  The list of calls must not be empty
+     *  The list of calls is executed in order
+     *
+     * @param _multicallSigner Address of an approved signer
+     * @param _reference Reference
+     * @param _calls List of calls
+     * @param _deadline Expiration timestamp
+     * @param _signature Signature of the multicall signer
+     */
+    // slither-disable-start low-level-calls,cyclomatic-complexity
+    // solhint-disable-next-line code-complexity
+    function execute(
+        address _multicallSigner,
+        bytes32 _reference,
+        Call[] calldata _calls,
+        uint256 _deadline,
+        bytes calldata _signature
+    ) external nonReentrant {
+        // solhint-disable-next-line not-rely-on-time
+        if (_deadline < block.timestamp) {
+            revert Expired(_deadline);
+        }
+        if (_reference == 0) {
+            revert InvalidReference(_reference);
+        }
+        if (replayProtection[_reference]) {
+            revert ReusedReference(_reference);
+        }
+        if (_calls.length == 0) {
+            revert EmptyCallArray();
+        }
+        for (uint256 i = 0; i < _calls.length; i++) {
+            if (bytes(_calls[i].functionSignature).length == 0) {
```

Comment:
Code coverage report indicates this branch isn't tested

Reply:
Note that I used the following, and then searched for GuardedMulticaller2:
forge coverage --report debug > x.txt

Reply:
I found the `debug` report not reliable. I added a test that should cover that line: 
https://github.com/immutable/contracts/pull/241/files#diff-7891bd472e64eb648ea5f38d97917eacbfc82723eeb5beffe2c65ee58c4c6b30R268

```
calls[0] = GuardedMulticaller2.Call(address(target), "", abi.encodePacked(uint256(42)));
```